sirepo/pkcli/job_agent.py

```diff
@@ -55,14 +57,48 @@ def default_command():
     pkdlog('{}', cfg)
     i = tornado.ioloop.IOLoop.current()
     d = _Dispatcher()
-    def s(n, x):
-        return i.add_callback_from_signal(d.terminate)
+    def s(*args):
+        return i.add_callback_from_signal(_terminate, d)
     signal.signal(signal.SIGTERM, s)
     signal.signal(signal.SIGINT, s)
     i.spawn_callback(d.loop)
     i.start()
 
 
+def start_sbatch():
+    def kill_agent(pid_file):
+        if socket.getfqdn() == pid_file.fqdn:
+            os.kill(pid_file.pid, signal.SIGKILL)
+        else:
+            try:
+                subprocess.run(
+                    ('ssh', pid_file.fqdn, 'kill', '-KILL', str(pid_file.pid)),
+                    capture_output=True,
+                ).check_returncode()
+            except subprocess.CalledProcessError as e:
+                pkdp(
+                    'cmd={cmd} returncode={returncode} stderr={stderr}',
+                    **vars(e)
+                )
+    try:
+        kill_agent(pkjson.load_any(pkio.py_path(_PID_FILE)))
+    except Exception as e:
+        if pkio.exception_is_not_found(e):
+            pass
+        pkdlog('error={} stack={}', e, pkdexc())
```

Comment:
Only log the stack and error if not found. You probably don't want to log not founds at all, since that's the normal case.

I think the read should happen before the kill_agent call. This is the "caveat" that not found could be thrown in kill_agent for another reason (process does not exist) so don't catch that exception in the same place.